tls/tls_srv.c

```diff
@@ -239,15 +232,10 @@ ttls_parse_supported_point_formats(TlsCtx *tls, const unsigned char *buf,
 	}
 
 	tls->hs->cli_exts = 1;
-	for (list_size = buf[0], p = buf + 1; list_size > 0; --list_size, ++p) {
-		if (p[0] == TTLS_ECP_PF_UNCOMPRESSED
-		    || p[0] == TTLS_ECP_PF_COMPRESSED)
-		{
-			tls->hs->ecdh_ctx.point_format = p[0];
-			T_DBG("ClientHello: point format selected: %d\n", p[0]);
-			return 0;
-		}
-	}
+	/*
+	 * Uncompressed is the only point format supported by RFC 8422,
+	 * so just ignore the list.
+	 */
```

Comment:
The control flow of this function looks wrong — returned code is always zero regardless of the actual buffer contents. It doesn't matter now in our fork when all cases were stripped. I get it.
But in the mbedtls code they process the buffer, and if no supported point format was found, return success leaving `.point_format` uninitialized. I wonder why?

Reply:
All the memory is zeroized in mbed TLS, so technically the point format is equal to `MBEDTLS_ECP_PF_UNCOMPRESSED`. Probably, the constant is zero to be the default value.